core/app/models/spree/order_update_attributes.rb

```diff
@@ -0,0 +1,42 @@
+module Spree
+  class OrderUpdateAttributes
+    # @param order [Order] existing (persisted) order
+    # @param attributes [Hash] attributes which are assigned to the new order.
+    #   These attributes should already have been filtered.
+    #   * :payments_attributes attributes
+    def initialize(order, attributes, request_env: nil)
+      @order = order
+      @attributes = attributes.dup
+      @payments_attributes = @attributes.delete(:payments_attributes) || []
+      @request_env = request_env
+    end
+
+    # Assign the attributes to the order and save the order
+    # @return true if saved, otherwise false and errors will be set on the order
+    def apply
+      assign_order_attributes
+      assign_payments_attributes
+
+      if order.save
+        order.set_shipments_cost if order.shipments.any?
+        true
+      else
+        false
+      end
+    end
+
+    private
+
+    attr_reader :attributes, :payments_attributes, :order
+
+    def assign_order_attributes
+      order.assign_attributes attributes
```

Comment:
Can we just leave this in `apply`? I don't see the point of extracting one line of code that we only call once.